src/containers/Cinema/Cinema.tsx

```diff
@@ -1,192 +1,139 @@
-    };
-
-    const initializePlayer = () => {
-      if (!window.jwplayer || !playerElementRef.current) return;
-
-      playerRef.current = window.jwplayer(playerElementRef.current) as JWPlayer;
-
-      playerRef.current.setup({
-        playlist: [deepCopy(item)],
-        aspect: false,
-        width: '100%',
-        height: '100%',
-        mute: false,
-        autostart: true,
-        repeat: false,
-      });
-
-      attachEvents();
-      calculateWatchHistoryProgress();
-      setPlayer(playerRef.current);
-
-      playerRef.current.setPlaylistItemCallback(handlePlaylistItemCallback);
-    };
-
-    if (playerRef.current) {
-      return loadPlaylist();
-    }
-
-    if (libLoaded) {
-      initializePlayer();
-    }
-  }, [libLoaded, item, enableWatchHistory, setPlayer, handlePlaylistItemCallback, detachEvents, attachEvents, player]);
+  const handleUserActive = useCallback(() => setUserActive(true), []);
+  const handleUserInactive = useCallback(() => setUserActive(false), []);
+  const handlePlaylistItemCallback = usePlaylistItemCallback();
 
+  // effects
   useEffect(() => {
-    return () => {
-      if (playerRef.current) {
-        // Detaching events before component unmount
-        detachEvents();
-        playerRef.current.remove();
-      }
-    };
-  }, [detachEvents]);
+    if (open) setUserActive(true);
+    if (!open) saveItem(item, getProgress());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [open]);
 
   return (
-    <div className={classNames(styles.cinema, { [styles.fill]: !isTrailer })}>
-      <div ref={playerElementRef} />
-    </div>
+    <Fade open={open}>
+      <div className={styles.playerContainer}>
+        <div className={styles.player}>
+          <Player
+            playerId={player}
+            feedId={feedId}
+            item={item}
+            onReady={handleReady}
+            onPlay={handlePlay}
+            onPause={handlePause}
+            onComplete={handleComplete}
+            onUserActive={handleUserActive}
+            onUserInActive={handleUserInactive}
+            onPlaylistItemCallback={handlePlaylistItemCallback}
+            startTime={startTime}
+          />
+        </div>
+        <Fade open={!isPlaying || userActive}>
+          <div className={styles.playerOverlay}>
+            <div className={styles.playerContent}>
+              <IconButton aria-label={t('common:back')} onClick={onClose} className={styles.backButton}>
+                <ArrowLeft />
+              </IconButton>
+              <div>
+                <h2 className={styles.title}>{title}</h2>
+                <div className={styles.metaContainer}>
+                  {isSeries && (
+                    <div className={styles.seriesMeta}>
+                      <strong>{seriesMeta}</strong> - {episodeTitle}
+                    </div>
+                  )}
+                  <div className={styles.meta}>{videoMeta}</div>
+                </div>
```

Comment:
This could also be more generic e.g. `firstMetaString` and `secondMetaString`

Reply:
What about `primaryMetadata` and `secondaryMetadata`?